From 8015776f8cc57cedb750829366c7898ac8697768 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 04 Sep 2020 12:31:40 +0300
Subject: [PATCH] [SW-12754] gaudi: security: add sram cfg protect

Implement SRAM CFG protection for all banks using protection bits.
SRAM CFG RTRs cannot be protected by protection bits due to a VLSI
integration version mismatch bug.

Change-Id: I5259f19ca940c6a6d9c2b75e688874cb42f32ce7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index da64035..e662da4 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -820,6 +820,24 @@
 	preboot_secure_prot_nics_engs_regs(reg, mask, 1);
 }
 
+#define SRAM_BANKS_NUM	32
+#define SRAM_BANKS_DIST	0x8000
+static void preboot_secure_sram_cfg(void)
+{
+	u32 base;
+
+	base = mmSRAM_Y0_X0_BANK_BASE & ~HL_CFG_BASE;
+	set_prot_bits_pages_range_all(base, SRAM_BANKS_DIST, SRAM_BANKS_NUM);
+
+	/*
+	 * TODO: There is an integration version mismatch bug in Gaudi
+	 * resulting in absence of the Privilege protection bits registers.
+	 * Therefore we cannot use the protection bits to protect the
+	 * SRAM_Y_X_RTRs.
+	 * LBW RRs should be used instead.
+	 */
+}
+
 static void preboot_security_pci_disable(void)
 {
 	regmap_writel(0, mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MIN_0);
@@ -873,6 +891,7 @@
 	preboot_secure_pll();
 	preboot_secure_nic();
 	preboot_secure_if();
+	preboot_secure_sram_cfg();
 
 	/* TODO: implement the rest of security enablement */
 
